crates/c-api/src/store.rs

```diff
@@ -203,6 +204,28 @@ pub extern "C" fn wasmtime_context_set_wasi(
     })
 }
 
+#[cfg(feature = "wasi")]
+#[no_mangle]
+pub extern "C" fn wasmtime_context_set_default_wasi_if_not_exist(
```

Comment:
Theoretically no, but practically it enables a cleaner way to guarantee a valid wasi will be returned even if the caller is not calling SetWasi.

Reply:
Ok in that case I'd prefer to avoid adding this because the C API isn't really one geared towards ergonomics (it is C after all), but such wrappers could always be present in higher-level languages like Go for example.

Reply:
Would you like to suggest a specific change here? Like, tracking if `SetWasi` is ever called in the caller?

Reply:
Yes I don't think this API should be added at this time. If a caller wants to conditionally enable WASI or not then I think that should be tracked by the caller for now.